src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -637,6 +637,19 @@ void AbstractWriteLog<I>::shut_down(Context *on_finish) {
       }
       update_image_cache_state(next_ctx);
     });
+  ctx = new LambdaContext(
+    [this, ctx](int r) {
+      ldout(m_image_ctx.cct, 6) << "complete in flight operations" << dendl;
+      m_work_queue.queue(ctx, r);
+    });
```

Comment:
I mean instead of adding this, why not use 
```
next_ctx = util::create_async_context_callback(m_work_queue, next_ctx);
```

Reply:
`m_work_queue` is not `WQ *work_queue` type, it is `ContextWQ m_work_queue`, compile failed.

Reply:
`m_work_queue` is not `WQ *work_queue` type, it is `ContextWQ m_work_queue`, compile failed.

Reply:
Sorry, sure it should have been `util::create_async_context_callback(&m_work_queue, next_ctx)`.

Reply:
Done. Thank you for your guidance. I'm not familiar with this usage and data structure. I learned it.